src/iterable_recursive_open_struct.rb

```diff
@@ -0,0 +1,26 @@
+
+module Metalware
+  class IterableRecursiveOpenStruct < RecursiveOpenStruct
+    def each(&block)
+      convert_hash_values_to_own_class.each(&block)
+    end
+
+    def each=(*args)
+      raise IterableRecursiveOpenStructPropertyError,
+        "Cannot set property 'each', reserved to use for iteration"
+    end
+
+    private
+
+    def convert_hash_values_to_own_class
```

Comment:
But would happen if you tried to call `convert_hash_values_to_own_class` from inside `IterableRecursiveOpenStruct`?

Would it still call the private method? or the newly defined value of `convert_hash_values_to_own_class` and return foo?